# $\vec{B}$ amplitude scaling, How to modify $(\alpha_{B1}, \alpha_{B2})$ to achieve the same mobility $\mu$

Let's say for a given amplitude $B$ a convenient couple $(\alpha_{B1}, \alpha_{B2})$ achieves good stationary state.
If we scale $B$ with a factor $a \in \mathbb{R}, a > 0$, what is the $\alpha_B$ scaling factor $b \in \mathbb{R}, b > 0$ such that $\mu ' = \mu$ ?

In other words:
* Let $a \in \mathbb{R}, a > 0$. $B' \equiv a B$.
* What is $b \in \mathbb{R}, b > 0$ used in $\alpha_B ' = b \alpha_B$ such that $\mu ' = \mu$ ?

We recall that:
$$
\mu \equiv \frac{e}{m_e \nu_m}\frac{1}{1 + \frac{\omega_{ce}^2}{\nu_m^2}} 
$$
With $\nu_m = \alpha_B \omega_{ce}$. Indeed, we consider $K_{el} = \nu_{ew} = 0$ here.

So:
$$ \mu ' = \mu  \Leftrightarrow \frac{e}{m_e a b \alpha_B \omega_{ce}}\frac{1}{1 + \frac{a^2 \omega_{ce}^2}{a^2 b^2 \alpha_B^2 \omega_{ce}^2}} = \frac{e}{m_e \alpha_B \omega_{ce}}\frac{1}{1 + \frac{\omega_{ce}^2}{\alpha_B^2 \omega_{ce}^2}} $$
$$\Leftrightarrow ab(1 + \frac{1}{b^2 \alpha_B^2}) = 1 + \frac{1}{\alpha_B^2}$$
$$\Leftrightarrow a \alpha_B^2 b^2 - (\alpha_B^2 + 1)b + a = 0$$

Which is a polynomial in $b$ of second order. So the similar scaling admits a real solution iff:
$$\Delta \equiv (\alpha_B^2 + 1)^2 - 4 a^2 \alpha_B^2 \geq 0$$
$$\Leftrightarrow \alpha_B^2 + 1 - 2 a \alpha_B \geq 0$$
$$\Leftrightarrow a \leq \frac{\alpha_B^2 + 1}{2 \alpha_B}$$

But in facts, I always choose anomalous coefficients in this range $ 0 < \alpha_B < 0.1 $ so that $5 < \frac{\alpha_B^2 + 1}{2 \alpha_B}$. So if $a \leq 5$(and with this usual $\alpha_B$ range) there is a real solution $b$ achieving a similar scaling, equivalent of being the solution of:
$$ a \alpha_B^2 b^2 - (\alpha_B^2 + 1)b + a = 0 $$
$$\Leftrightarrow b_{\pm} = \frac{\alpha_B^2 + 1 \pm \sqrt{(\alpha_B^2 + 1)^2 - 4 a^2 \alpha_B^2}}{2 a \alpha_B^2}$$

Both solutions $b_{\pm} > 0$ and thus are admittable.

Now the question is for typical values and coming from a nominal couple $(\alpha_{B1}, \alpha_{B2})$.
Nominal case $B_{max} = 100$ G and the chosen couple is $(1.0826\times 10^{-2}, 2.0325\times 10^{-2})$.

We want to modify the $B$ array from $B_{max} = 100$ G maximum to $B_{max} = 50$ G so a scaling of $a = 0.5$.
According to the previous formulae:
$$b_{1 +} = \frac{\alpha_{B1}^2 + 1 + \sqrt{(\alpha_{B1}^2 + 1)^2 - 4 a^2 \alpha_{B1}^2}}{2 a \alpha_{B1}^2} = 1.706602\times 10^4$$
$$b_{1 -} = \frac{\alpha_{B1}^2 + 1 + \sqrt{(\alpha_{B1}^2 + 1)^2 - 4 a^2 \alpha_{B1}^2}}{2 a \alpha_{B1}^2} = 4.999561\times 10^{-1}$$
$$b_{2 +} = \frac{\alpha_{B2}^2 + 1 + \sqrt{(\alpha_{B2}^2 + 1)^2 - 4 a^2 \alpha_{B2}^2}}{2 a \alpha_{B2}^2} = 4.842877\times 10^3$$
$$b_{2 -} = \frac{\alpha_{B2}^2 + 1 + \sqrt{(\alpha_{B2}^2 + 1)^2 - 4 a^2 \alpha_{B2}^2}}{2 a \alpha_{B2}^2} = 4.998451\times 10^{-1}$$

So that:
$$b_{1 +} \alpha_{B1} = 1.8476\times 10^2 $$
$$b_{1 -} \alpha_{B1} = 5.4125\times 10^{-3} $$
$$b_{2 +} \alpha_{B2} = 9.8431\times 10^{1} $$
$$b_{2 -} \alpha_{B2} = 1.0159\times 10^{-2} $$

In [45]:
import math

## $\alpha_{B1}$

In [46]:
#Inputs
a = 2.0
alphaB = 1.0826e-2

In [47]:
D = (alphaB**2 + 1)**2 - 4 * a**2 * alphaB**2
bplus = ( (alphaB**2 + 1) + math.sqrt(D) )/(2 * a * alphaB**2)
bminus =( (alphaB**2 + 1) - math.sqrt(D) )/(2 * a * alphaB**2)  

In [48]:
print(f"b+ = {bplus:.6e}")
print(f"b- = {bminus:.6e}")


b+ = 4.264628e+03
b- = 2.000704e+00


In [49]:
print(f"b+*alphaB = {bplus*alphaB:.4e}")
print(f"b-*alphaB = {bminus*alphaB:.4e}")

b+*alphaB = 4.6169e+01
b-*alphaB = 2.1660e-02


## $\alpha_{B2}$

In [50]:
#Inputs
a = 2.0
alphaB = 2.0325e-2

In [51]:
D = (alphaB**2 + 1)**2 - 4 * a**2 * alphaB**2
bplus = ( (alphaB**2 + 1) + math.sqrt(D) )/(2 * a * alphaB**2)
bminus =( (alphaB**2 + 1) - math.sqrt(D) )/(2 * a * alphaB**2)  

In [52]:
print(f"b+ = {bplus:.6e}")
print(f"b- = {bminus:.6e}")


b+ = 1.208842e+03
b- = 2.002486e+00


In [53]:
print(f"b+*alphaB = {bplus*alphaB:.4e}")
print(f"b-*alphaB = {bminus*alphaB:.4e}")

b+*alphaB = 2.4570e+01
b-*alphaB = 4.0701e-02


## Similar parameters achieving $\bar{\mu} = 5.460 \times 10^{4}$ $cm^2$/(V.s) and steady-state $J_d = 690.0$  $A m^{-2}$. 

|$B_{max}$ [G]  |$\alpha_{B1}$              | $\alpha_{B2}$                 |
| -------       | ---------------------     | ----------------------        |
|   50          | $5.1425\times 10^{-3}$    | $1.0159\times 10^{-2}$        |
|   100         | $1.0826\times 10^{-2}$    | $2.0325\times 10^{-2}$        |
|   150         | $1.6241\times 10^{-2}$    | $3.0503\times 10^{-2}$        |
|   200         | $2.1660\times 10^{-2}$    | $4.0701\times 10^{-2}$        |